### DISCUSSION TITLE
Expose public port

### DISCUSSION THREAD

**Discussion by Thijmen** [coollabsio/coolify#2576] [Improvement Requests]
Hi there,

For services it's an option to make it "publicly available". It then spins up an nginx proxy which proxies the port to the internal container.

I wonder why this method has been chosen and not restarting the service with that port specifically exposed to the host. In my opinion, that makes the most sense. However, I may forget about a situation where this is not possible.

What is the reason behind this decision?

**Comment by ladderschool** [comment]
I noticed this as well and I'm also curious. Trying to expose a redis service spins up another container with nginx and I can't reach my redis instance over the internet.

**Reply by Thijmen** [reply]
While I can understand that a nginx service is quick (namely not having to restart the service), I'd rather rely on exposing the actual service to be honest.

**Comment by shafamail** [comment]
@Thijmen You can achieve this using Port mapping. 

You do not have to use the "make publicly available" as "Port mapping" also exposes the host port to the public if that is what you want.

It maps the container port to the desired host port. 

https://coolify.io/docs/knowledge-base/databases#ports-mapping-vs-public-port
